src/components/MessagingWindow/MessagingWindow.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import "./MessagingWindow.css";
+import Firebase from "firebase";
+import PropTypes from "prop-types";
+import Message from "./Message";
+
+class MessagingWindow extends Component {
+  constructor(props) {
+    super(props);
+
+    const { user, receiver } = this.props;
+    this.state = {
+      messages: [],
+      user,
+      receiver,
+      text: "" // temporary local variable used to handle text from the "send-message" text field
+    };
+  }
+
+  componentDidMount() {
+    const { user, receiver } = this.state;
+    this.tempRef = Firebase.database().ref(
+      `users/${user}/friends/${receiver}/`
+    ); // temporary reference to extract conversation reference
+    this.tempRef.on("value", snapshot1 => {

```

Comment:
Why are you listening twice? If you're trying to just get the conversation id once, you should use the once method [here](https://firebase.google.com/docs/database/web/read-and-write#read_data_once). Then once you have this conversation id, you can listen to the conversation branch.

I don't think you should be listening to the friends branch unless you are listening for new friends or new conversations.